core/src/main/java/cucumber/runtime/Runtime.java

```diff
@@ -65,4 +80,112 @@ public byte exitStatus() {
         return exitStatus.exitStatus(runtimeOptions.isStrict());
     }
 
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+
+        private EventBus eventBus = new TestCaseSyncEventBus(new DefaultEventBus(TimeService.SYSTEM));
+        private ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        private RuntimeOptions runtimeOptions = new RuntimeOptions("");
+        private BackendSupplier backendSupplier;
+        private ResourceLoader resourceLoader;
+        private ClassFinder classFinder;
+        private GlueSupplier glueSupplier = new RuntimeGlueSupplier();
+        private FeatureSupplier featureSupplier;
+        private List<Plugin> additionalPlugins = Collections.emptyList();
+
+        private Builder() {
+        }
+
+        public Builder withArg(final String arg) {
+            this.runtimeOptions = new RuntimeOptions(arg);
+            return this;
+        }
+
+        public Builder withArgs(final String... args) {
+            return withArgs(Arrays.asList(args));
+        }
+
+        public Builder withArgs(final List<String> args) {
+            this.runtimeOptions = new RuntimeOptions(args);
+            return this;
+        }
+
+        public Builder withRuntimeOptions(final RuntimeOptions runtimeOptions) {
+            this.runtimeOptions = runtimeOptions;
+            return this;
+        }
+
+        public Builder withClassLoader(final ClassLoader classLoader) {
+            this.classLoader = classLoader;
+            return this;
+        }
+
+        public Builder withResourceLoader(final ResourceLoader resourceLoader) {
+            this.resourceLoader = resourceLoader;
+            return this;
+        }
+
+        public Builder withClassFinder(final ClassFinder classFinder) {
+            this.classFinder = classFinder;
+            return this;
+        }
+
+        public Builder withBackendSupplier(final BackendSupplier backendSupplier) {
+            this.backendSupplier = backendSupplier;
+            return this;
+        }
+
+        public Builder withGlueSupplier(final GlueSupplier glueSupplier) {
+            this.glueSupplier = glueSupplier;
+            return this;
+        }
+
+        public Builder withFeatureSupplier(final FeatureSupplier featureSupplier) {
+            this.featureSupplier = featureSupplier;
+            return this;
+        }
+
+        public Builder withAdditionalPlugins(final Plugin... plugins) {
+            this.additionalPlugins = Arrays.asList(plugins);
+            return this;
+        }
+
+        public Builder withEventBus(final EventBus eventBus) {
+            this.eventBus = eventBus;
+            return this;
+        }
+
+        public Runtime build() {
```

Comment:
This is a really good idea.

Reply:
Thanks, I love the builder pattern - so helpful